### PR TITLE
refactor(core): :art: change CoreChip structure and fix samples

### PR DESCRIPTION
**Pull request by anantakumarghosh**
## Description

change CoreChip document structure and fix samples, 3 unable to fix in this PR:

- Sizes CoreChip
- Multiline CoreChip
- Array CoreChip

ref: #172
## Checklist

- [ ] I have performed a thorough self-review of my code.
- [ ] I have added or updated relevant tests for my changes.
- [ ] My code follows the project's style guidelines and best practices.
- [ ] I have updated the documentation if necessary.
- [ ] I have added or updated relevant comments in my code.
- [ ] I have resolved any merge conflicts of my branch.

## Reviewers

---

## Maintainer Notes

- [ ] Has this change been tested in a staging environment?
- [ ] Does this change introduce any breaking changes or deprecations?

### CODE REVIEW

File: app/components/component-docs/dataDisplay/CoreChip.docs.js [hunk 1]
```diff
@@ -30,306 +31,543 @@ function handleDelete() {
...
+              </SubtitleDescription></>}
+            code={`<CoreChip label="Clickable Link" component="a" href="#basic-chip" clickable />
+<CoreChip
+  label="Clickable Link"
+  component="a"
+  href="#basic-chip"
+  variant="outlined"
+  clickable
+/>`}
+            expandedCode={""}
```
Comment (techoneel) [inline app/components/component-docs/dataDisplay/CoreChip.docs.js:275]: why it is empty???

File: app/components/component-docs/dataDisplay/CoreChip.docs.js [hunk 2]
```diff
@@ -30,306 +31,543 @@ function handleDelete() {
...
+  deleteIcon={<CoreIcon icon="done" />}
+/>
+<CoreChip
+  label="Custom delete icon"
+  onClick={handleClick}
+  onDelete={handleDelete}
+  deleteIcon={<CoreIcon icon="delete" />}
+  variant="outlined"
+/>`}
+            expandedCode={`import { CoreChip, CoreStack, CoreIcon } from "@wrappid/core";
```
Comment (techoneel) [inline app/components/component-docs/dataDisplay/CoreChip.docs.js:315]: coding documentation also needs to be beautifully crafted.
```
    expandedCode={`
import { CoreChip, CoreStack, CoreIcon } from "@wrappid/core";
function handleClick() {
  console.info("CoreChip clicked! 🎉");
}
function handleDelete() {
  console.info("Delete icon clicked! 🗑️");
}
export default function CustomDeleteIconCoreChip() {
  return (
    <>
      <CoreStack direction="row" spacing={1}>
        <CoreChip
          label="Custom delete icon"
          onClick={handleClick}
          onDelete={handleDelete}
          deleteIcon={<CoreIcon icon="done" />}
        />
        <CoreChip
          label="Custom delete icon"
          onClick={handleClick}
          onDelete={handleDelete}
          deleteIcon={<CoreIcon icon="delete" />}
          variant="outlined"
        />
      </CoreStack>
    </>
  );
}
    `}
```

File: app/components/component-docs/dataDisplay/CoreChip.docs.js [hunk 3]
```diff
@@ -1,20 +1,21 @@
...
+  CorePaper
 } from "@wrappid/core";
 
-import CodeImport from "../../CodeImport";
+import CodeBlock from "../../CodeBlock";
 import CodeSample from "../../CodeSample";
-import ComponentProps from "../../ComponentProps";
+import ComponentDocs from "../ComponentDocs";
+const CORE_COMPONENT = CoreChip;
+const CORE_COMPONENT_NAME = (CoreChip?.displayName || CoreChip?.name);
```
Comment (anantakumarghosh) [inline app/components/component-docs/dataDisplay/CoreChip.docs.js:18]: In next commit,  `CoreChip?.displayName` will be changed to CORE_COMPONENT?.displayName

**Review by techoneel — DISMISSED** [review]
Please make the changes and re-request for review

File: app/components/component-docs/dataDisplay/CoreChip.docs.js (hunk 1 above)
Comment (anantakumarghosh) [reply app/components/component-docs/dataDisplay/CoreChip.docs.js:275]: As mentioned in the title, **NOT WORKING**.

CodeSample will be completed once the component works.

File: app/components/component-docs/dataDisplay/CoreChip.docs.js (hunk 2 above)
Comment (anantakumarghosh) [reply app/components/component-docs/dataDisplay/CoreChip.docs.js:315]: > coding documentation also needs to be beautifully crafted.
> 
> ```
>     expandedCode={`
> import { CoreChip, CoreStack, CoreIcon } from "@wrappid/core";
> function handleClick() {
>   console.info("CoreChip clicked! 🎉");
> }
> function handleDelete() {
>   console.info("Delete icon clicked! 🗑️");
> }
> export default function CustomDeleteIconCoreChip() {
>   return (
>     <>
>       <CoreStack direction="row" spacing={1}>
>         <CoreChip
>           label="Custom delete icon"
>           onClick={handleClick}
>           onDelete={handleDelete}
>           deleteIcon={<CoreIcon icon="done" />}
>         />
>         <CoreChip
>           label="Custom delete icon"
>           onClick={handleClick}
>           onDelete={handleDelete}
>           deleteIcon={<CoreIcon icon="delete" />}
>           variant="outlined"
>         />
>       </CoreStack>
>     </>
>   );
> }
>     `}
> ```

It is already beautifully crafted. Suggested change will add an extra line on top.

**Review by techoneel — APPROVED** [review]
good to go

**Comment by anantakumarghosh** [comment]
> Please make the changes and re-request for review

no changes required according to requested changes.

**Pull request merged**